File: SCSDPOngoing/src/Review/thirdPractice/CreationalPatterns.scala
package Review.thirdPractice

sealed trait Shape2 {
  
  def draw() : Unit
  
}

case class Circle2() extends Shape2 {
  
  def draw() : Unit = {
    println("I am drawing a circle")
  }
  
}

case class Square2() extends Shape2 {
  
  def draw() : Unit = {
    println("I am drawing an Square")
  }
  
}

case class Rectangle2() extends Shape2 {
  
  def draw() : Unit = {
    println("I am drawing a Rectangle")
  }
  
}

object ShapeFactory {
  
  def getShape(name : String) : Shape2 =
    name match {
      case "circle" => Circle2()
      case "square" => Square2()
      case "Rectangle" => Rectangle2()
      case _ => null;
  }
  
}


object CreationalPatterns extends App {
  
  val shape = ShapeFactory.getShape("circle")
  shape.draw()
  
  
}

File: SDPOngoing/src/Revision/CreationalPatterns.java
package Revision;

public class CreationalPatterns {
	
	interface Shape {
		
		public void draw();
	}
	
	class Square implements Shape {
		
		@Override
		public void draw(){
			System.out.println("Drawing an Square");
		}
	}
	
	class Triangle implements Shape {
		
		@Override
		public void draw(){
			System.out.println("Drawing a Triangle");
		}
	}
	
	class Circle implements Shape {
		
		@Override
		public void draw(){
			System.out.println("Drawing a Circle");
		}
	}
	
	public class ShapeFactory {
		
		public Shape getShape (String name) {
			Shape result = null;
			switch (name) {
			case "Square" : result = new Square();
			break;                
			case "Circle" : result = new Circle();
			break;
			}
			return result;
		}
		
		
	}
	
	
	public void launch(){
		
        Singleton.getInstance();
		
		ShapeFactory factory = new ShapeFactory();
		
		Shape test = factory.getShape("Square");
		
		Shape test2 = new Square();
		test.draw();
		
		
	}
	
	

	public static void main(String[] args) {
		
		Singleton.getInstance();
		
		CreationalPatterns script = new CreationalPatterns();
		script.launch();
		
		//ShapeFactory factory = new ShapeFactory();
		
		
		
		// TODO Auto-generated method stub

	}

}
